x/xion/keeper/keeper.go

```diff
@@ -62,3 +74,63 @@ func (k Keeper) OverwritePlatformPercentage(ctx sdktypes.Context, percentage uin
 func (k Keeper) GetAuthority() string {
 	return k.authority
 }
+
+// DispatchActions attempts to execute the provided messages via authorization
+// grants from the message signer to the grantee.
+func (k Keeper) DispatchActions(ctx sdk.Context, grantee sdk.AccAddress, msgs []sdk.Msg) ([][]byte, error) {
```

Comment:
I would approach this differently. Since you are basically requiring this endpoint to only accept cosmwasm Authz messages, the user will need to know which type of grant needs to go to which msg handler, which is complicated. 

Also, this handler only handles the CodeID logic, and can't expand past that. We want something more general.

I think a more expandable approach is to create a new type of Authorization, something like a KeeperAuthorization or StatefulAuthorization, that implements the standard authorization interface but returns an error on Accept and adds a new function like StatefulAccept that accepts the keeper as an input.

I would basically take the existing DispatchActions logic from the authz keeper, and copy it here. Then, after the GetAuthorization call, I would check to see if the grant implements this new interface, if so call the new Accept method, otherwise call the normal one from the authz interface